Python/Strings/1276.py

```diff
@@ -0,0 +1,56 @@
+alfabeto = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
+
+while True:
+    try:
+				# Leitura da entrada com remoção dos espaços vazios
+        texto = input().replace(' ', '')
+        letras = []
+
+				# Para cada posição da array texto é inserido seu valor na array 'letra', exceto se este valor já tiver sido inserido
+        for letra in texto:
+            try:
+								# Procura se já esta no vetor letras
+                letras.index(letra)
+            except ValueError:
+								# Caso não esteja é inserido
+                letras.insert(len(letras), letra)
+        
+        faixas = []
+				
+				# Organização do vetor em ordem alfabética
+        letras = sorted(letras)
+        primeiraLetra = ''
+        ultimaLetra = ''
+        
+        for letra in letras:
+						# No primeiro loop atualiza o valor das variáveis 'primeiraLetra' e 'ultimaLetra'
+            if primeiraLetra == '':
+                primeiraLetra = letra
+                ultimaLetra = letra
+            else:
+								# letra sempe está um valor a frete de 'ultimaletra', então caso sua posição no alfabeto for igual à posição 'ultimaletra'+1, significa que está em ordem alfabética, e portanto 'ultimaletra' é atualizado
+                if alfabeto.index(letra) == alfabeto.index(ultimaLetra) + 1:
+                    ultimaLetra = letra
+                else:
+										# Caso não esteja em ordem alfabética, a faixa de letras é inserido no vetor 'faixas' 
+                    faixas.insert(len(faixas), primeiraLetra + ':' + ultimaLetra)
+                    primeiraLetra = letra
+                    ultimaLetra = letra
+
+				# Inserindo os últimos valores no vetor						
+        if primeiraLetra and ultimaLetra:
+            faixas.insert(len(faixas), primeiraLetra + ':' + ultimaLetra)
+        elif primeiraLetra and ultimaLetra == '':
+            faixas.insert(len(faixas), primeiraLetra + ':' + ultimaLetra)
+
+				
+        textoFinal = ''
+
+				# Armazena os valores das faixas em 'textoFinal'
+        for faixa in faixas:
+            textoFinal += faixa + ', '
+					
+        # Imprime o texto final sem ', ' final
+        print(textoFinal[0: len(textoFinal) - 2])
+    except EOFError:
+        break
```

Comment:
```suggestion
# Problema 1276 - Beecrowd - Strings - Nível 3

while True:
    try:
        # Leitura da entrada com remoção dos espaços vazios
        texto = input().replace(' ', '')
        
        # Vamos converter o texto em um conjunto, para eliminar as letras repetidas
        letras = set(texto)

        # Vamos converter o conjunto para uma lista ordenada
        letras = sorted(list(letras))

        # Se a entrada for vazia apenas printamos uma linha vazia
        if len(letras) == 0:
            print()
            continue

        # Para poder verificar ordem, vamos converter as letras em números
        numeros = list(map(ord, letras))

        # Definimos o inicío e o fim da sequência como o primeiro número
        inicio = numeros[0]
        fim = numeros[0]

        # Lista vazia de faixas
        faixas = []

        # Para cada número vemos se ele é a continuação do fim
        for n in numeros[1:]:
            # Se sim, incrementamos o fim
            if n == fim + 1:
                fim += 1
            # Se não, terminamos a faiza corrente e iniciamos uma nova com o novo valor
            else:
                faixas.append((inicio, fim))
                inicio = fim = n

        # Precisamos levar em conta a última faixa
        faixas.append((inicio, fim))

        # Como o padrão de saída requer que as faizas sejam separadas por virgula
        # Vamos converter cada faixa para a notação usada: a:b
        faixas = map(lambda f : f'{chr(f[0])}:{chr(f[1])}', faixas)
        
        # Então printar as faixar unidas por virgula
        print(', '.join(faixas))

    except EOFError:
        break
```

Reply:
Você não aproveitou alguns padrões interessante do python:

- set() ao invés de letras = [] e um for com if e try catch
- ord(), chr() e map() ao invés de uma lista alfabeto, definida pelo usuário, procurando o índice da letra nessa lista para fazer comparação a cada iteração
- append() ao invés de insert(n, ...), o padrão para inserção em lista no python é o append, que coloca no fim, é literalmente o equivalente a lista.insert(len(lista)) só que faz tudo de uma vez, é mais eficiente
- join() ao invés de toda a última parte do código, você desenvolve toda uma lógica só para construir a string final, loop para adicionar as virgulas e slicing para remover as mesmas do fim, tudo isso é resolvido com um ', '.join(lista)